frontend/src/APIClients/AuthAPIClient.ts

```diff
@@ -23,23 +28,40 @@ const login = async (
   }
 };
 
+type SignUpFunction = (
```

Comment:
I think this can be simplified to
```
type SignUpFunction = (
  options: MutationFunctionOptions<
    { signup: AuthenticatedUser },
    OperationVariables
  >,
) => Promise<FetchResult<{ signup: AuthenticatedUser }>>;
```
((Test the code after making this change just to be sure))